monai/apps/auto3dseg/ensemble_builder.py

```diff
@@ -582,9 +582,18 @@ def ensemble(self):
         builder.set_ensemble_method(self.ensemble_method)
         self.ensembler = builder.get_ensemble()
         infer_files = self.ensembler.infer_files
-        infer_files = partition_dataset(
-            data=infer_files, shuffle=False, num_partitions=self.world_size, even_divisible=True
-        )[self.rank]
+        if len(infer_files) < self.world_size:
+            if len(infer_files) == 0:
+                logger.info("No existing infer files. Ensembler ending.")
```

Comment:
consider changing test "No testing files for inference is provided, ensure testing key exists in the datalist json file"

Reply:
ok